qiskit/pulse/schedule.py

```diff
@@ -62,7 +62,8 @@ def __init__(self, *schedules: List[Union[ScheduleComponent, Tuple[int, Schedule
         try:
             self._timeslots = TimeslotCollection(*timeslots)
         except PulseError as ts_err:
-            raise PulseError('Child schedules {0} overlap.'.format(schedules)) from ts_err
+            raise PulseError('Schedules overlap: {0} for {1}'
+                             ''.format(ts_err.message, schedules)) from ts_err
```

Comment:
Given my comment below on `__repr__`, rather than printing the repr of the schedules, maybe print something like you had below for the two schedules which are causing the issue (rather than their str representation). Ideally, we'd show the instructions which overlapped, but I realize that the current structure with `timeslots` makes that challenging, so we can skip it.

Reply:
looks like we have no information of what exact two schedules gives us overlap due to
```python
self._timeslots = TimeslotCollection(*timeslots)
```

I see 3 options (I'm not fully familiar with code as you may notice, that is why only 3 :) ):
1. Leave message as `'Schedules overlap: {0} for {1}'''.format(ts_err.message, schedules)` which will tell us which two timeslots are overlapping and will print all schedules.
2. Write message `'Scheduler overlap {}'.format(schedules)` which will give us all schedules and we can get information regarding how they are overlaped down the stacktrace by Timeslots error. 
3. We can rewrite `__init__` method of `Schedule` to merge `TimesplotsCollection` pairwise and not all `*timeslots`, something like:
```python
        timeslots = TimeslotCollection()
        _children = []
        for sched_pair in schedules:
            # recreate as sequence starting at 0.
            if not isinstance(sched_pair, (list, tuple)):
                sched_pair = (0, sched_pair)
            # convert to tuple
            sched_pair = tuple(sched_pair)
            insert_time, sched = sched_pair
            sched_timeslots = sched.timeslots
            if insert_time:
                sched_timeslots = sched_timeslots.shift(insert_time)
            _children.append(sched_pair)

            try:
                timeslots = TimeslotCollection(timeslots, sched_timeslots)
            except PulseError as ts_err:
                raise PulseError('Schedule {0} overlap existing timeslots {1}'
                                 ''.format(sched, timeslots)) from ts_err
```
but it will potentially have worst performance

Reply:
No worries, let's keep it simple and keep the current behavior of printing all schedules, then. But rather than printing the whole schedule (the default `__repr__`) print minimal info for each one (like "Schedule(name={}, t=[{}, {}])").

Reply:
Perhaps we could try the third quickly and test if it hurts performance much? We can tolerate a small hit to performance for enhanced useability.

Reply:
@taalexander  @lcapelluto 

wrote a quick test on above mentioned implementation and compared original and new impl:
```python
N_TESTS = 100

results = {}
for n_schedules in range(10, 1000, 50):
    runtimes = []
    for _ in range(N_TESTS):
        start = datetime.datetime.now()
        schedules = []
        for idx in range(n_schedules):
            s = pulse.Schedule()
            s |= pulse.Acquire(10)(qubits=system.qubits[0], mem_slots=system.memoryslots[0]) << idx*10
            schedules.append(s)
        merged = pulse.Schedule(*schedules)
        end = datetime.datetime.now()
        runtime = end - start
        runtimes.append(runtime.microseconds)

    results[n_schedules] = float(sum(runtimes)) / N_TESTS
```

and results are:
<a href="https://ibb.co/ScQf1B1"><img src="https://i.ibb.co/ScQf1B1/Screen-Shot-2019-11-04-at-5-04-32-PM.png" alt="Screen-Shot-2019-11-04-at-5-04-32-PM" border="0"></a>

which is in avg 10% slower than the old one.

I believe it worth it to rewrite init function. Error became readable with `__repr__` with full information and you know which schedule cause error. 
On top of that to format them nicely like `"Schedule(name={}, t=[{}, {}])"` we have to iterate through list and figure out instance type, due to `__init__` method of schedule accepts list of schedules or list of tuples, which will make raise statement bulky :) 

I already feel how @lcapelluto hates me for such a long conversations in such a small issue :)

Reply:
We love all the contributions and all the long conversations!! I'm excited about getting this issue merged. Sorry for the delay, I'm looking now :)

Reply:
To give us the full picture of this tradeoff, can you please post an example error message for each of the simple implementation and the implementation above? Thank you @IceKhan13!

Reply:
1. for `'Schedules overlap: {0} for {1}'.format(ts_err.message, schedules)`:
```qiskit.pulse.exceptions.PulseError: 'Schedules overlap: Overlap on channel MemorySlot(0) over time range [5, 10] for (Schedule(5 (5, Instruction(Acquire(acq1, duration=100, kernel=None, discriminator=None), AcquireChannel(1), MemorySlot(0)))), Schedule(0 (0, Instruction(Acquire(acq0, duration=10, kernel=None, discriminator=None), AcquireChannel(0), MemorySlot(0)))))'```

2. for `'Scheduler overlap {}'.format(schedules)`:
```qiskit.pulse.exceptions.PulseError: 'Scheduler overlap (Schedule(5 (5, Instruction(Acquire(acq1, duration=100, kernel=None, discriminator=None), AcquireChannel(1), MemorySlot(0)))), Schedule(0 (0, Instruction(Acquire(acq0, duration=10, kernel=None, discriminator=None), AcquireChannel(0), MemorySlot(0)))))'```

3. for third implementation:
```qiskit.pulse.exceptions.PulseError: 'Schedule(0 (0, Instruction(Acquire(acq0, duration=10, kernel=None, discriminator=None), AcquireChannel(0), MemorySlot(0)))) overlaps existing timeslots TimeslotCollection{AcquireChannel(1): [(5, 105)], MemorySlot(0): [(5, 105)]}'```

Reply:
I think either 1 or 2 is good. What do you think @lcapelluto?

Reply:
I like 1.
Rather than printing the schedule default str, could we try the sort of format you had earlier used in repr? I think it would be like `['Schedule(name="{}", duration={})'.format(sch.name, sch.duration) for sch in schedules]`. (I find it a little difficult to parse all the instructions in line). This will be so helpful :)))

Reply:
Done @ https://github.com/Qiskit/qiskit-terra/pull/3319/commits/a45cdaf7bd1e2433049466715838b8b311f533df

Now it looks like:
```Schedules overlap: Overlap on channel MemorySlot(0) over time range [5, 10] for Schedule(name=“acq0”, duration=10), Schedule(name=“acq0", duration=15)```